db/migrate/20171023011304_migrate_data_to_vendor.rb

```diff
@@ -0,0 +1,18 @@
+class MigrateDataToVendor < ActiveRecord::Migration
+  def up
+
+    Item.all.each do |item|
+      v = Vendor.where(
+        name: item.source
+      ).first_or_create
+
+      item.vendor_id = v.id
```

Comment:
I think you need to save the `item` here so the change persists?

Reply:
ok!

Reply:
Sorry I should've been more clear: try something like `item.update_columns(vendor_id: v.id)` to skip callbacks and validations, in addition to being faster it is likely to be safer than:

```ruby
item.vendor_id = v.id
item.save
```

in this example, if a validation fails it will silently discard the update which we probably don't want; once the `source` column is removed we permanently lose the data